driver/gl/window.go

```diff
@@ -503,50 +593,104 @@ func keyToName(key glfw.Key) fyne.KeyName {
-	if w.canvas.Focused() == nil && w.canvas.onTypedKey == nil {
-		return
-	}
+
 	if action != glfw.Press { // ignore key up
+		// TODO: handle repeated key
 		return
 	}
 
-	if key <= glfw.KeyWorld1 { // filter printable characters handled in charModInput
+	keyName := keyToName(key)
+	keyDesktopModifier := desktopModifier(mods)
+
+	if keyDesktopModifier < desktop.ControlModifier {
+		// No shortcut detected, pass down to TypedKey and exit
+		keyEvent := &fyne.KeyEvent{Name: keyName}
+		if w.canvas.Focused() != nil {
+			go w.canvas.Focused().TypedKey(keyEvent)
+		}
+		if w.canvas.onTypedKey != nil {
+			go w.canvas.onTypedKey(keyEvent)
+		}
 		return
 	}
 
-	ev := new(fyne.KeyEvent)
-	ev.Name = keyToName(key)
+	isMac := runtime.GOOS == "darwin"
 
-	if w.canvas.Focused() != nil {
-		go w.canvas.Focused().TypedKey(ev)
+	var shortcut fyne.Shortcuter
+	switch keyName {
+	case fyne.KeyV:
+		// detect paste shortcut
+		if (!isMac && keyDesktopModifier == desktop.ControlModifier) || (isMac && keyDesktopModifier == desktop.SuperModifier) {
+			shortcut = &fyne.ShortcutPaste{
+				Clipboard: w.Clipboard(),
+			}
+		}
+	case fyne.KeyC:
+		// detect copy shortcut
+		if (!isMac && keyDesktopModifier == desktop.ControlModifier) || (isMac && keyDesktopModifier == desktop.SuperModifier) {
+			shortcut = &fyne.ShortcutCopy{
+				Clipboard: w.Clipboard(),
+			}
+		}
+	case fyne.KeyX:
+		// detect cut shortcut
+		if (!isMac && keyDesktopModifier == desktop.ControlModifier) || (isMac && keyDesktopModifier == desktop.SuperModifier) {
+			shortcut = &fyne.ShortcutCut{
+				Clipboard: w.Clipboard(),
+			}
+		}
+	default:
+		shortcut = &desktop.CustomShortcut{
+			KeyName:  keyName,
+			Modifier: keyDesktopModifier,
+		}
+	}
+
+	if shortcutable, ok := w.canvas.Focused().(fyne.Shortcutable); ok {
+		go shortcutable.HandleShortcut(shortcut)
+	}
+
+	go w.canvas.HandleShortcut(shortcut)
+}
+
+func desktopModifier(mods glfw.ModifierKey) desktop.Modifier {
+	var m desktop.Modifier
+	if (mods & glfw.ModShift) != 0 {
+		m |= desktop.ShiftModifier
+	}
+	if (mods & glfw.ModControl) != 0 {
+		m |= desktop.ControlModifier
+	}
+	if (mods & glfw.ModAlt) != 0 {
+		m |= desktop.AltModifier
 	}
-	if w.canvas.onTypedKey != nil {
-		go w.canvas.onTypedKey(ev)
+	if (mods & glfw.ModSuper) != 0 {
+		m |= desktop.SuperModifier
 	}
+	return m
 }
 
+// charModInput defines the character with modifiers callback which is called when a
+// Unicode character is input regardless of what modifier keys are used.
+//
+// The character with modifiers callback is intended for implementing custom
+// Unicode character input. Characters do not map 1:1 to physical keys,
+// as a key may produce zero, one or more characters.
 func (w *window) charModInput(viewport *glfw.Window, char rune, mods glfw.ModifierKey) {
 	if w.canvas.Focused() == nil && w.canvas.onTypedRune == nil {
 		return
 	}
 
-	if mods == 0 || mods == glfw.ModShift {
```

Comment:
Is this no longer needed to guard against reporting runes where a modifier was present?
I am pretty sure that if you press "Alt"+"F" then as well as the shortcut handler it will report a rune unless this is guarded...